src/puck.ts

```diff
@@ -347,49 +410,78 @@ export class LookupPuck {
     window.addEventListener('pointercancel', this.stopDraggingPuck);
   };
 
-  private readonly stopDraggingPuck = () => {
-    this.isBeingDragged = false;
+  private readonly onPuckSingleClick = () => {
+    // TODO: toggle whether the puck is enabled or disabled.
+  };
+
+  private readonly onPuckDoubleClick = () => {
+    this.targetOrientation =
+      this.targetOrientation === 'above' ? 'below' : 'above';
+    this.setPositionWithinSafeArea(this.puckX, this.puckY);
+  };
+
+  // May be called manually (without an event), or upon 'pointerup' or 'pointercancel'.
+  private readonly stopDraggingPuck = (event?: PointerEvent) => {
     if (this.puck) {
       this.puck.style.pointerEvents = 'revert';
       this.puck.classList.remove('dragging');
-
-      // Update the target orientation if the puck was parked low down on the
-      // screen.
-      const { viewportHeight } = this.getViewportDimensions(
-        this.puck?.ownerDocument || document
-      );
-
-      const { bottom: safeAreaBottom } =
-        this.safeAreaProvider.getSafeArea() || {
-          top: 0,
-          right: 0,
-          bottom: 0,
-          left: 0,
-        };
-
-      // The distance from the bottom of the earth (which can only travel within
-      // the safe area) to the centre of the moon (which is the point from which
-      // the mouse events are fired). This is effectively the height of the
-      // "blind spot" that a puck supporting only the "above" orientation would
-      // have.
-      const activePuckVerticalExtent =
-        this.earthHeight +
-        (this.targetAbsoluteOffsetYAbove - this.earthHeight / 2);
-      this.targetOrientation =
-        this.puckY >= viewportHeight - safeAreaBottom - activePuckVerticalExtent
-          ? 'below'
-          : 'above';
       this.setPositionWithinSafeArea(this.puckX, this.puckY);
     }
 
     window.removeEventListener('pointermove', this.onWindowPointerMove);
     window.removeEventListener('pointerup', this.stopDraggingPuck);
     window.removeEventListener('pointercancel', this.stopDraggingPuck);
+
+    if (!event) {
+      if (this.clickStateHasTimeout(this.clickState)) {
+        window.clearTimeout(this.clickState.timeout);
+        this.clickState = { kind: 'idle' };
+      }
+      return;
+    }
+
+    const targetIsPuckOrRoot =
+      event.target === this.puck || event.target === this.container;
+    if (event.type === 'pointercancel' || !targetIsPuckOrRoot) {
+      // Stop tracking this click and wait for the next 'pointerdown' to come along instead.
+      if (this.clickStateHasTimeout(this.clickState)) {
+        window.clearTimeout(this.clickState.timeout);
+      }
+      this.clickState = { kind: 'idle' };
+    } else if (event.type === 'pointerup' && targetIsPuckOrRoot) {
```

Comment:
When testing this on Firefox the double click behaviour wasn't working for me because of the check for `targetIsPuckOrRoot`.

Specifically, because we set `pointerEvents` to `'none'` in `onPuckPointerDown` the `event.target` we get back in Firefox at least is the `<body>` element and hence this check fails and we never run the double-click behaviour.

Reply:
(Incidentally, if we _don't_ need this check after all we won't need to hoist `container` up to a class member either.)

Reply:
Resolved by your PR. Whatever I found it necessary for before doesn't seem to be the case anymore (probably due to your `pointercapture` changes).